proto/buf/alpha/registry/v1alpha1/list.proto

```diff
@@ -0,0 +1,26 @@
+// Copyright 2020-2021 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package buf.alpha.registry.v1alpha1;
+
+// OrderedColumn defines the options available for specifying
+// the column by which the results should be ordered.
+enum OrderedColumn {
```

Comment:
I don't think we should have a shared enum for any listing endpoints as I imagine the sort options won't be the same across all endpoints. For example, sorting by owner column for ListUsers seems odd.

For a typed API, I think I would prefer the enums to be `{field}_ASC/DESC` options rather than having an additional `bool reversed`. If we ever allow the api to expand (sort by x first, then by y) you can express that.

Other nit: column to me is a database implementation detail and I would avoid putting it in

Reply:
+1 on RPC specific enums, sorry it wasn't made clear in my initial comment. I would argue that `reversed` is already well established in the API, so there's no need to remove the existing one, but the point about being able to order by two or more columns is a neat feature of combining them! I think we could still allow that in the future if we wanted to.

Also  +1 that these should be named based on fields on the resource they operate on, not leaking SQL jargon like column. ListUsers could have `UserSort` and `USER_SORT_USERNAME`, etc.

Reply:
thoughts here @johanbrandhorst @bufdev? i don't know if i think it makes sense to have a different sorting type enum for each entity we need to sort (users vs repositories vs plugins etc) - that would make me lean towards the original idea of simply a string for the ordering query instead of a bunch of different enums

and noted the comment about columns - def can make the adjustment to avoid in naming

Reply:
dangit my comment was too late - ok noted @johanbrandhorst, will revisit

Reply:
> I would argue that reversed is already well established in the API, so there's no need to remove the existing on

And similarly the established pattern in the api is that you cannot change the sort order, only reverse it. Now that we're exchanging that to allow specifying what to sort on, I think it makes sense to change into ascending/descending which are standard and have clear meaning, rather than continue to propagate `reversed` (which only has relative meaning compared to some default).

Reply:
That's totally fair 👍🏻